README.adoc

```diff
@@ -38,153 +40,199 @@ endif::[]
+We use GitHub Packages to distribute the SDK. In order to consume our dependencies you need to generate a GitHub Personal Access Token. Please follow the how to link:https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#authenticating-to-github-packages[authenticate to GitHub Packages, window="_blank"].
 
-Just open your terminal and enter:
+NOTICE:: You need to have _read_ access to this repository and generate a https://github.com/settings/tokens/new/[personal access token] with `repo` and `read:packages` scope.
 
-[source,bash]
-----
-curl -s "https://get.sdkman.io" | bash
-----
+===== Token
 
-Follow the setup instructions and verify that your installation works:
+The token needs to be made available.
 
-[source,bash]
-----
-sdk version
-----
+. Add `gpr.user = {GitHub username}` and `gpr.key = {GitHub Personal Access Token}` to your global Gradle properties `~/.gradle/gradle.properties`
 
-If it shows you something like `SDKMAN 5.9.0+555` then your good to continue.
+    gpr.user=github-username
+    gpr.key=github-token
 
-Just install the mentioned dependencies:
+. Or add following environment variables `PACKAGE_REGISTRY_USERNAME={GitHub username}` and `PACKAGE_REGISTRY_TOKEN={GitHub Personal Access Token}`
 
-[source,bash]
-----
-sdk install java 8.0.265-amzn
-sdk install kotlin 1.4.10
-----
+===== Setup Maven Repository
 
-Check your installation by:
+Add the following maven repository configuration to your root build.gradle:
 
-[source,bash]
+[source, Gradle]
 ----
-java -version
-kotlin -version
+    allprojects {
+        repositories {
+            ...
+            maven {
+                url = URI("https://maven.pkg.github.com/d4l-data4life/hc-fhir-sdk-kmp")
+                credentials {
+                    username = project.findProperty("gpr.user") as String? ?: System.getenv("PACKAGE_REGISTRY_USERNAME")
+                    password = project.findProperty("gpr.key") as String? ?: System.getenv("PACKAGE_REGISTRY_TOKEN")
+                }
+            }
+        }
+    }
 ----
 
-==== Install Python
+== Usage
 
-It is recommended to use link:https://github.com/pyenv/pyenv[PyEnv] for managing dedicated python versions.
+How to use this SDK.
 
-Use link:https://brew.sh/[Homebrew] to install it on mac:
+=== Serialization
 
-[source,bash]
-----
-brew install pyenv
+Create an instance of your desired `FhirStu3Parser` or `FhirR4Parser` using `FhirParserFactory.createStu3Parser()` or `FhirParserFactory.createR4Parser()`.
+
+Decoding and Encoding JSON FHIR data is as simple as:
+
+==== Decode
+
+In case you know the type of FHIR JSON data, you could directly decode it.
+
+[source, Kotlin]
 ----
+val parser = FhirParserFactory.createR4Parser()
 
-Make sure you are in the project root and install the correct Python version which is derived from `.python-version` file:
+val data = "JSON FHIR data"
 
-[source,bash]
+val result = parser.fromJson(DomainResource::class, "JSON FHIR data")
```

Comment:
I guess this should be `data`, right otherwise `val data =  "JSON FHIR data"` is redundant.